src/cssjanus.js

```diff
@@ -387,32 +387,41 @@ function CSSJanus() {
 
 cssjanus = new CSSJanus();
 
+if ( typeof window === 'object' ) {
```

Comment:
I haven't seen this particular way of exporting a module before. If this was based on something else, I'd appreciate a link so I can read more about it.

For Wikimedia projects, we usually follow the following pattern:
```js
  if ( typeof module !== 'undefined' && module.exports ) {
    module.exports = Thing;
  } else {
    this.Thing = Thing;
  }
```

Or, if there are additional members that should remain private:

```js
( function ( global ) {

  // ...

  if ( typeof module !== 'undefined' && module.exports ) {
    module.exports = Thing;
  } else {
    global.Thing = Thing;
  }
}( this  ) );
```

The closure would avoid exposing the other variables and classes to the browser scope, which seems undesirable.

I'm hoping Closure Compiler will know to not mangle the `transform` key of the object literal retuned from CSSJanus when it sees that it is exported publicly. I would prefer not to have quoted keys for Closure Compiler only.

If the quoting is needed, I wonder if it would work to quote the key directly on line 308. That would avoid the need for re-assignment here.

Reply:
I think using a quoted key for "transform" (line-disabling the linter) at the place its declared, would be a fine patch.

For the global variable we may still need a different patch, though. But I'd be fine with using a quoted index for that as well. Actually, I think we should just in general make the global variable export explicit. Right now there is no code for that because it's intended to run in Node.js.

Perhaps something like this?

```lang
if ( typeof module !== 'undefined' && module.exports ) {
  // ..
} else {
  this[ "cssjanus" ] = cssjanus;
}
```

And as part of that, also making the entire file privately scoped so that both with and without Closure Compiler, the other internal classes don't become global variables.

Adding an IFFY around a Node.js module seems odd though, so we may want to do that as part of a pre-publish build step for use in local development and in the npm-published package (something like dist/cssjanus.js for browsers).